pkg/kubelet/dockertools/exec.go

```diff
@@ -96,10 +94,16 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		err = command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
 	}
-
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
```

Comment:
I much prefer the way @sttts wrote it. It's unambiguous and you don't have to reason through it.